trillian/migrillian/election/etcd/election.go

```diff
@@ -76,27 +56,18 @@ func (e *Election) Await(ctx context.Context) (context.Context, error) {
 		cancel()
 		return nil, ctx.Err()
 	case rsp, ok := <-ch:
-		if !ok {
-			cancel()
-			return nil, errMasterUnconfirmed
-		}
-		if string(rsp.Kvs[0].Value) != e.instanceID {
+		if !ok || string(rsp.Kvs[0].Value) != e.instanceID {
+			// Mastership has been overtaken in the meantime, or not capturead at all.
 			cancel()
-			return nil, errMasterOvertaken
+			return cctx, nil
 		}
 	}
 
 	// At this point we have observed confirmation that we are the master; start
 	// a goroutine to monitor for anyone else overtaking us.
-	done := make(chan struct{})
 	go func() {
-		defer func() {
-			glog.Infof("%d: canceling mastership context", e.treeID)
-			// Note: close comes before context cancelation, so that Await can be
-			// invoked immediately after the cancelation without an error.
-			close(done)
-			cancel()
-		}()
+		defer cancel()
+		defer glog.Infof("%d: canceling mastership context", e.treeID)
```

Comment:
Why drop the `defer func() { ... }()` wrapper, btw?  (I had to think for a moment about Go's `defer` ordering...)

Reply:
My thinking was something like:

1. It's shorter.
2. I'd better protect against `Infof` panicking (not sure if it can), and cancel always.

Well, 1. shouldn't be a stopper, 2. I could perhaps reword `Infof` and put it after cancel. Changed the code accordingly. What do you think? Maybe removing `Infof` completely would be even better?